CONTRIBUTING.md

```diff
@@ -0,0 +1,71 @@
+# Contributing to AccessKit
+
+## Making a Pull Request
+
+### Cargo.lock File
+
+AccessKit intentionally includes the `Cargo.lock` file in our git repository.
+This is mainly due to our bindings to other languages.
+
+Usually you should not run cargo update when creating a pull request even when adding/changing a dependency.
+Simply building the library should update the `Cargo.lock` file with the minimal changes needed.
+
+> [!NOTE]
+> This is not normal / best practice for most libraries.
+> If you are writing your own library, your `Cargo.lock` file is not used when the library is included in another project.
+> This means your CI will be testing with an out of date set of dependencies compared to your users, missing potential problems.
+> See the [official documentation](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries) for more information. 
+
+### CHANGELOG.md
+
+Our CHANGELOG.md files are auto generated using [Release Please](https://github.com/googleapis/release-please) and should not be edited manually.
+
+To control what is in the CHANGELOG.md for your change, use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) when writing your commit messages.
+
+#### Example Commit Messages
+Taken from [Conventional Commits documentation](https://www.conventionalcommits.org/en/v1.0.0/#summary):
+> feat: allow provided config object to extend other configs
+>
+> BREAKING CHANGE: `extends` key in config file is now used for extending other config files
+
+> fix: prevent racing of requests
+>
+> Introduce a request id and a reference to latest request. Dismiss
+incoming responses other than from latest request.
```

Comment:
I haven't used Release Please before. You might want different examples based on what you want to encourage use of.

Can also just remove the examples if you think the link to Conventional Commits is enough, I figured it might be nice to not have to leave the page to get a basic idea of it was my reasoning.